### PR TITLE
[DPN-24/LocalStorage] update to master, so far 1 test for each functi…

### PR DESCRIPTION
**Pull request by Yifan-Zhang618**
…on, will add test later and refactor format later

### CODE REVIEW

**Comment by trevorliberty** [comment]
That failing check is actually an interesting problem. We may need to approach that copy link feature another way. There is no copying mechanism on a headless version of chrome:

https://stackoverflow.com/questions/53080757/python-3-selenium-clipboard-not-working-on-headless-chromedriver-on-windows

https://stackoverflow.com/questions/59409183/python-selenium-clipboard-functionality-ctrl-c-not-working-in-firefox-headl

Apparently we can use a workaround with phantomjs, but this seems like an awful lot of overhead. We may want to ignore this in the effort of getting this PR through

**Comment by jjesse55** [comment]
@trevorliberty @Yifan-Zhang618 In the sake of time, I agree. However instead of ignoring the failures I would say comment out test(s) for now, create a new issue for finding the fix to this, and assign it to Yifan since he's already tried a few things.

File: selenium_tests/test_storage.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        super().__init__(methods)
+        self.chrome_options = Options()
+        self.chrome_options.add_argument("--headless")
+        self.driver = webdriver.Chrome(chrome_options=self.chrome_options)
+
+    def setUp(self):
+        self.driver.get(URL)
+
+    def tearDown(self):
+        self.driver.quit()
```
Comment (jjesse55) [inline selenium_tests/test_storage.py:26]: - Inherit from the SeleniumTestBase
- remove last three lines of constructor
- remove setup, teardown methods

File: selenium_tests/test_storage.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+            data = json.load(f)
+            for item in data:
+                self.assertEqual(item['value'], test_value)
+
+        os.remove(path + '/parameters.json')
+
+    def test_if_copy_link_successfully(self):
+        """ Test that Copy link icon correctly copied current url """
+        self.driver.find_element_by_id("copy-link").click()
+        time.sleep(2)
```
Comment (jjesse55) [inline selenium_tests/test_storage.py:84]: Just curious, does it not copy in time without a call to sleep?

File: requirements.txt [hunk 3]
```diff
@@ -10,3 +10,5 @@ sqlparse==0.4.1
 sympy==1.7
 scipy==1.6.0
 urllib3==1.26.3
+clipboard==0.0.4
+pyperclip==1.8.1
```
Comment (jjesse55) [inline requirements.txt:14]: Are you still using this dependency or was this left over from debugging the copy/paste testing issue?

**Review by jjesse55 — CHANGES_REQUESTED** [review]

File: requirements.txt (hunk 3 above)
Comment (jjesse55) [reply requirements.txt:14]: Pyperclip, that is

**Review by jjesse55 — APPROVED** [review]

File: requirements.txt (hunk 3 above)
Comment (jjesse55) [reply requirements.txt]: @Yifan-Zhang618 is this dependency still needed?

**Review by trevorliberty — APPROVED** [review]

**Pull request merged**